File: web/admin/src/types/validations/shipping.ts
import type { ValidationArgs } from '@vuelidate/core'
import { maxLengthArray, maxLength, maxValue, minLengthArray, minValue, required } from '~/lib/validations'

export const ShippingValidationRules: ValidationArgs = {
  name: { required, maxLength: maxLength(64) },
  hasFreeShipping: {},
  box60Frozen: { required, minValue: minValue(0), maxValue: maxValue(9999999999) },
  box80Frozen: { required, minValue: minValue(0), maxValue: maxValue(9999999999) },
  box100Frozen: { required, minValue: minValue(0), maxValue: maxValue(9999999999) },
}

export const ShippingRateValidationRules: ValidationArgs = {
  name: { required, maxLength: maxLength(64) },
  price: { required, minValue: minValue(0), maxValue: maxValue(9999999999) },
  prefectureCodes: { minLengthArray: minLengthArray(1), maxLengthArray: maxLengthArray(47) },
}

export const CreateShippingValidationRules: ValidationArgs = {
  name: { required, maxLength: maxLength(64) },
  hasFreeShipping: {},
  freeShippingRates: { required, minValue: minValue(0), maxValue: maxValue(9999999999) },
  box60Frozen: { required, minValue: minValue(0), maxValue: maxValue(9999999999) },
  box80Frozen: { required, minValue: minValue(0), maxValue: maxValue(9999999999) },
  box100Frozen: { required, minValue: minValue(0), maxValue: maxValue(9999999999) },
}

export const UpdateShippingValidationRules: ValidationArgs = {
  name: { required, maxLength: maxLength(64) },
  hasFreeShipping: {},
  box60Frozen: { required, minValue: minValue(0), maxValue: maxValue(9999999999) },
  box80Frozen: { required, minValue: minValue(0), maxValue: maxValue(9999999999) },
  box100Frozen: { required, minValue: minValue(0), maxValue: maxValue(9999999999) },
}

export const UpdateDefaultShippingValidationRules: ValidationArgs = {
  hasFreeShipping: {},
  box60Frozen: { required, minValue: minValue(0), maxValue: maxValue(9999999999) },
  box80Frozen: { required, minValue: minValue(0), maxValue: maxValue(9999999999) },
  box100Frozen: { required, minValue: minValue(0), maxValue: maxValue(9999999999) },
}

File: api/internal/gateway/user/v1/response/shipping.go
package response

// Shipping - 配送設定情報
type Shipping struct {
	ID                string          `json:"id"`                // 配送設定ID
	Box60Rates        []*ShippingRate `json:"box60Rates"`        // 箱サイズ60の通常（常温・冷蔵便）配送料一覧
	Box60Frozen       int64           `json:"box60Frozen"`       // 箱サイズ60の追加（冷凍便）追加配送料(税込)
	Box80Rates        []*ShippingRate `json:"box80Rates"`        // 箱サイズ80の通常（常温・冷蔵便）配送料一覧
	Box80Frozen       int64           `json:"box80Frozen"`       // 箱サイズ80の追加（冷凍便）追加配送料(税込)
	Box100Rates       []*ShippingRate `json:"box100Rates"`       // 箱サイズ100の通常（常温・冷蔵便）配送料一覧
	Box100Frozen      int64           `json:"box100Frozen"`      // 箱サイズ100の追加（冷凍便）追加配送料(税込)
	HasFreeShipping   bool            `json:"hasFreeShipping"`   // 送料無料オプションの有無
	FreeShippingRates int64           `json:"freeShippingRates"` // 送料無料になる金額(税込)
}

type ShippingRate struct {
	Number          int64    `json:"number"`          // No.
	Name            string   `json:"name"`            // 配送料金設定名
	Price           int64    `json:"price"`           // 配送料金(税込)
	Prefectures     []string `json:"prefectures"`     // 対象都道府県名
	PrefectureCodes []int32  `json:"prefectureCodes"` // 対象都道府県一覧
}
